plugins/expired_gpg_keys.py

```diff
@@ -0,0 +1,105 @@
+import dnf
+import rpm
+import subprocess
+
+from datetime import datetime
+from dnfpluginscore import _, logger
+
+
+class ExpiredGPGKeys(dnf.Plugin):
+    """
+    Find expired GPG keys and suggest their removal.
+
+    This is a workaround to solve https://github.com/rpm-software-management/dnf/issues/2075.
+    """
+
+    name = 'expired-gpg-keys'
+
+    def __init__(self, base, cli):
+        super(ExpiredGPGKeys, self).__init__(base, cli)
+        self.base = base
+        self.cli = cli
+
+    def config(self):
+        if not self.base.conf.gpgcheck:
+            return
+
+        if not self.is_gpg_installed():
+            return
+
+        for (hdr, expire_date) in self.list_expired_keys():
+            print(_("The following GPG key has expired on {0}:".format(expire_date)))
+            print("    {0}\n".format(hdr["summary"]))
+            print(_("For more information about the key:"))
+            print("    rpm -qi {0}\n".format(hdr[rpm.RPMTAG_NVR]))
+
+            print(_("Installing packages signed with this key will fail due to a GPG check error.\n"
+                    "It is recommended to remove the expired key to allow the installation of\n"
+                    "packages signed with an updated key."))
```

Comment:
I would write:

  "Installing packages signed with this key will fail because of it.\n"
  "It is recommended to remove the expired key to allow importing\n"
  "an updated key."

It's closer to what actually happens and what the user can expect. Consider that user will see this question for any expired key regardless he's going to install a packages signed with it, or install packages at all.

Reply:
Thanks, I've bit modified the message, but the result should be very similar.